src/doc/ndarray_for_numpy_users/mod.rs

```diff
@@ -524,6 +524,30 @@
 //! </td></tr>
 //! </table>
 //!
+//! ## Type conversions
+//!
+//! In `ndarray`, conversions between datatypes are done with `mapv()` by
+//! passing a closure to convert every element independently.
+//! For the conversion itself, we have several options:
+//! - `std::convert::From` ensures perfect, safe conversions at compile-time and
+//!   is generally recommended.
+//! - `std::convert::TryFrom` can be used for potentially unsafe conversions. It
+//!   will panic if values are encountered at runtime that cannot be converted
+//!   perfectly.
+//! - The `as` keyword performs a *saturating* cast since Rust 1.45 and should
+//!   only be used for safe, clear conversions such as upcasting.
+//! For details, be sure to check out the type conversion examples.
+//!
+//! NumPy | `ndarray` | Notes
+//! ------|-----------|------
+//! `a.astype(np.float32)` | `a.mapv(\|x\| f32::from(x))` | convert `u8` array infallibly safe to `f32` array with `std::convert::From`, generally recommended
+//! `a.astype(np.uint8)` | `a.mapv(\|x\| u8::try_from(x))` | try to convert `i8` array to `u8` array, panic if any value cannot be converted perfectly at runtime (e.g. negative value)
```

Comment:
I think this would result in an array of `Result`s as the intended `.unwrap()` is missing?

Reply:
Yes absolutely, thank you!